File: sample-java/src/main/java/com/jetradarmobile/sample_java/MainActivity.java
package com.jetradarmobile.sample_java;


import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.jetradarmobile.sociallogin.SocialLogin;
import com.jetradarmobile.sociallogin.SocialLoginCallback;
import com.jetradarmobile.sociallogin.SocialNetwork;
import com.jetradarmobile.sociallogin.SocialToken;
import com.jetradarmobile.sociallogin_facebook.FacebookNetwork;

import org.jetbrains.annotations.NotNull;

public class MainActivity extends AppCompatActivity implements
		View.OnClickListener,
		SocialLoginCallback {

	private TextView info;

	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);

		info = (TextView) findViewById(R.id.info);

		Button facebookLoginButton = (Button) findViewById(R.id.facebookButton);
		facebookLoginButton.setOnClickListener(this);
	}

	@Override
	public void onClick(View v) {
		switch (v.getId()) {
			case R.id.facebookButton:
				SocialLogin.Factory.with(this).loginTo(new FacebookNetwork(), this);
				break;
			default:
				break;
		}
	}

	@Override
	public void onLoginSuccess(@NotNull SocialNetwork socialNetwork,
	                           @NotNull SocialToken socialToken) {
		displayLoginInfo(socialToken);
	}

	@Override
	public void onLoginError(@NotNull SocialNetwork socialNetwork,
	                         @NotNull String errorMessage) {
		displayError(errorMessage);
	}

	private void displayLoginInfo(SocialToken token) {
		String info = "token = " + token.getToken() + "\n\n" +
				"user id = " + token.getUserId() + "\n\n" +
				"user name = " + token.getUserName() + "\n\n";

		this.info.setText(info);
	}

	private void displayError(String error) {
		info.setText(error);
	}
}

File: sample-kotlin/src/main/java/com/jetradarmobile/sociallogin/sample/MainActivity.kt
package com.jetradarmobile.sociallogin.sample

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.jetradarmobile.sociallogin.SocialToken
import com.jetradarmobile.sociallogin_facebook.FacebookNetwork
import com.jetradarmobile.sociallogin_google.GoogleNetwork
import com.jetradarmobile.sociallogin_odnoklassniki.OkNetwork
import com.jetradarmobile.sociallogin_rx.RxSocialLogin
import com.jetradarmobile.sociallogin_twitter.TwitterNetwork
import com.jetradarmobile.sociallogin_vkontakte.VkNetwork
import io.reactivex.Observable
import kotlinx.android.synthetic.main.activity_main.*
import ru.ok.android.sdk.util.OkScope

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        facebookButton.setOnClickListener {
            val scope = mutableListOf<String>()
            scope.add("public_profile")
            scope.add("email")

            subscribe(RxSocialLogin.instance.loginTo(this, FacebookNetwork(scope)))
        }

        vkontakteButton.setOnClickListener {
            val scope = mutableListOf<String>()
            scope.add("access_token")
            scope.add("email")

            subscribe(RxSocialLogin.instance.loginTo(this, VkNetwork(scope)))
        }

        twitterButton.setOnClickListener {
            subscribe(RxSocialLogin.instance.loginTo(this, TwitterNetwork(
                    getString(R.string.twitter_app_id),
                    getString(R.string.twitter_app_secret))))
        }

        odnoklassnikiButton.setOnClickListener {
            val scope = mutableListOf<String>()
            scope.add(OkScope.VALUABLE_ACCESS)
            scope.add(OkScope.LONG_ACCESS_TOKEN)

            subscribe(RxSocialLogin.instance.loginTo(this, OkNetwork(
                    getString(R.string.ok_app_id),
                    getString(R.string.ok_app_public_key),
                    getString(R.string.ok_redirect_url),
                    scope)))
        }

        googleButton.setOnClickListener {
            subscribe(RxSocialLogin.instance.loginTo(this, GoogleNetwork(
                    getString(R.string.google_server_client_id)
            )))
        }
    }

    private fun subscribe(observable: Observable<SocialToken>) {
        observable.subscribe(
                { token -> showText(getTokenText(token)) },
                { error -> showText(error.message ?: "error") })
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        RxSocialLogin.instance.onActivityResult(requestCode, resultCode, data)
        super.onActivityResult(requestCode, resultCode, data)
    }

    private fun showText(text: String) {
        info.text = text
    }

    private fun getTokenText(token: SocialToken) = "token = ${token.token} \n\n" +
            "user id = ${token.userId} \n\n" +
            "user name = ${token.userName} \n\n" +
            "email = ${token.email} \n\n"
}
